### PR TITLE
CDH-27606 Fix Morphlines.ReadRCFile to read null values in RCFile

### PR DESCRIPTION
**Pull request by prazanna**
Test case failed with the same error without the fix and passes with the fix.

### CODE REVIEW

File: kite-morphlines/kite-morphlines-hadoop-rcfile/src/test/java/org/kitesdk/morphline/hadoop/rcfile/ReadRCFileTest.java [hunk 1]
```diff
@@ -101,8 +102,8 @@ public void testRCFileRowWise() throws Exception {
     startSession();
     assertEquals(1, collector.getNumStartEvents());
     assertTrue(morphline.process(input));
-    assertTrue(areFieldsEqual(expected, collector.getRecords(), NUM_COLUMNS,
-        true));
+    assertTrue(
```
Comment (madrob) [inline kite-morphlines/kite-morphlines-hadoop-rcfile/src/test/java/org/kitesdk/morphline/hadoop/rcfile/ReadRCFileTest.java]: Please avoid unrelated formatting changes.

File: kite-morphlines/kite-morphlines-hadoop-rcfile/src/test/java/org/kitesdk/morphline/hadoop/rcfile/ReadRCFileTest.java [hunk 2]
```diff
@@ -135,7 +153,12 @@ private void createRCFile(final String fileName, final int numRecords,
       BytesRefArrayWritable dataWrite = new BytesRefArrayWritable(maxColumns);
       dataWrite.resetValid(maxColumns);
       for (int column = 0; column < maxColumns; column++) {
-        Text sampleText = new Text("ROW-NUM:" + row + ", COLUMN-NUM:" + column);
```
Comment (madrob) [inline kite-morphlines/kite-morphlines-hadoop-rcfile/src/test/java/org/kitesdk/morphline/hadoop/rcfile/ReadRCFileTest.java]: Can this logic be extracted into a method? It gets used in several places.

File: kite-morphlines/kite-morphlines-hadoop-rcfile/src/test/java/org/kitesdk/morphline/hadoop/rcfile/ReadRCFileTest.java [hunk 3]
```diff
@@ -122,8 +123,25 @@ public void testRCFileColumnWise() throws Exception {
         false));
   }
 
+  @Test
+  public void testRCFileWithNull() throws Exception {
```
Comment (madrob) [inline kite-morphlines/kite-morphlines-hadoop-rcfile/src/test/java/org/kitesdk/morphline/hadoop/rcfile/ReadRCFileTest.java]: This differs from `testRCFileRowWise` by only one line. Can we pull the common parts into a helper method?

**Comment by madrob** [comment]
The commit message should not reference a "CDH" issue, it should reference a CDK issue.

File: kite-morphlines/kite-morphlines-hadoop-rcfile/src/test/java/org/kitesdk/morphline/hadoop/rcfile/ReadRCFileTest.java [hunk 4]
```diff
@@ -135,7 +153,12 @@ private void createRCFile(final String fileName, final int numRecords,
       BytesRefArrayWritable dataWrite = new BytesRefArrayWritable(maxColumns);
       dataWrite.resetValid(maxColumns);
       for (int column = 0; column < maxColumns; column++) {
-        Text sampleText = new Text("ROW-NUM:" + row + ", COLUMN-NUM:" + column);
+        Writable sampleText = new Text(
+            "ROW-NUM:" + row + ", COLUMN-NUM:" + column);
+        // Set the last column of the last row as null
```
Comment (madrob) [inline kite-morphlines/kite-morphlines-hadoop-rcfile/src/test/java/org/kitesdk/morphline/hadoop/rcfile/ReadRCFileTest.java]: Is only setting the last element to null sufficient testing? What about null somewhere in the middle? What if the first element is null?

**Comment by whoschek** [comment]
Thanks for the fix, Prasanna!
